main/src/mill/modules/Jvm.scala

```diff
@@ -191,7 +190,7 @@ object Jvm {
     // to System.in/System.out/System.err, so the subprocess's streams get sent
     // to the parent process's origin outputs even if we want to direct them
     // elsewhere
-    if (System.in.isInstanceOf[ByteArrayInputStream]) {
+    if (System.in.isInstanceOf[PipedInputStream]) {
```

Comment:
I'm not hundred percent certain, that we are correct here. We probably want to keep `ByteArrayInputStream` as alternative proof for a "fake" or redirected input.

Reply:
I mean, we should keep both, `PipedInputStream` and `ByteArrayInputStream` (OR-ed) in the check.

Reply:
Yeah, I understand this.
My reasoning was the following: We use the `Jvm` API in many places, and it is especially also used to handle processes and in-process Java tools in Mill plugins. As such, the `runSubprocess` method isn't limited to its use as internal Mill. And it should work as expected when an external plugin uses it to handle some sub processes, As using `ByteArrayInputStream` is widely used to control the input of a spawned process, we should support it.
But I will not insist in that standpoint. Your solution is more lean and solves exactly one issue.

Reply:
I understand that `runSubprocess` is exposed to plugins and custom tasks and stuff, but all of those run in the context of a Mill build JVM? In that case, there won't be any `ByteArrayOutputStream` anymore, just a `PipedInputStream`.

If you were sub-processing out to a separate JVM you launched with the Mill classpath, and _then_ calling `Jvm.subProcess`, then i could imagine wanting to check for a dummy stdin. But AFAIK we never do that: Mill plugins and custom tasks are always in the Mill's JVM when they call `Jvm.subProcess`. `Jvm.subProcess` is also in the Mill `main` module, so it's not like some helper library we expect to use elsewhere.

Reply:
Nice explanation. Agreed.